admin_guide/http_proxies.adoc

```diff
@@ -0,0 +1,166 @@
+= Working with HTTP Proxies
+{product-author}
+{product-version}
+:icons: font
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+:description: This should be a clear and concise description of the topic.
+:keywords: comma, separated, keywords, list
+
+toc::[]
+
+== Overview
+Production environments can deny direct access to the web and instead have an
+HTTP or HTTPS proxy available. Configuring OpenShift builds and deployments to
+use these proxies can be as simple as setting standard environment variables in
+configuration or JSON files.
+
+[[configuring-hosts-for-proxies]]
+
+== Configuring Hosts for Proxies
+
+. Add the `*NO_PROXY*`, `*HTTP_PROXY*`, and `*HTTPS_PROXY*` environment variables
+to each host's *_/etc/sysconfig/openshift-master_* or
+*_/etc/sysconfig/openshift-node_* file depending on the type of host:
+
+====
+----
+HTTP_PROXY=http://USERNAME:PASSWORD@10.0.1.1:8080/
+HTTPS_PROXY=https://USERNAME:PASSWORD@10.0.0.1:8080/
+NO_PROXY=master.hostname.example.com
+----
+====
+
+. Restart the master or node as appropriate:
++
+----
+# systemctl restart openshift-master
+# systemctl restart openshift-node
+----
+
+[[proxying-docker-pull]]
+
+== Proxying Docker Pull
+
+Docker pulls content from the registry that hosts your built images. If you have
+an existing internal registry, use the proxy's IP with the NO_PROXY setting.
+This blacklists that registry's address, leaving the external http proxy as the
+only option.
+
```

Comment:
IMHO this paragraph is unclear. Suggest rewording to: "[Docker pulls... images.]. OpenShift nodes need to perform push and pull operations to registries. If you have a Docker registry that nodes should *not* access through a proxy make sure to include that registry's address in the NO_PROXY exclusion list"

Then I would update the example below to something like this:

NO_PROXY=master.hostname.example.com,172.30.123.45,docker-registry.default.svc.cluster.local

where 172.30.123.45 is the internal docker registry's service IP, and docker-registry.default.svc.cluster.local is the service name for a service called *docker-registry* in the *default* namespace (this is the default service name for the internal registry deployed by *oadm registry*)